### PR TITLE
Remove `rad app switch`

### LINKED ISSUE #6836: Remove `rad app switch`
## Overview of feature request

We have a CLI command `rad app switch` that can be used to set a 'current' application at the user/workspace level.

This command isn't really used by anyone and has already been replaced with a directory-based configuration system. It's much more intuitive to support the concept of 'current' application based on the directory structure than it is to encode it in user-wide config.

We should remove this. 

## Acceptance criteria

Remove `rad app switch` command

Remove plumbing for 'default' application in our configuration.

[AB#10464](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10464)

### PR DESCRIPTION
**Pull request by rynowak**
# Description

This change removes the `rad app switch` command. This command isn't really documented/advertised so we suspect it is not widely used. We also replaced this command with a newer directory-based way of configuring the current application that's more useful.

Removing this so we don't have to maintain it.

## Type of change

- This pull request adds or changes features of Radius and has an approved issue (issue link required).

Fixes: #6836

### Summary
🗑️🛠️🚚

Remove `rad app switch` command and related code. Simplify CLI validation and error messages to use application name from flags or directory.

> _`rad app switch` gone_
> _No more default application_
> _CLI is simpler_

### Walkthrough
*  Remove the `rad app switch` command and its references ([link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-d36b7b4d04f01a6cf60c50dd9c01da8fb8c190231ea4c59cbbfbe070178b510bL33), [link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-d36b7b4d04f01a6cf60c50dd9c01da8fb8c190231ea4c59cbbfbe070178b510bL287-L289), [link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-39764141ee6d5ec09b84841112602fa4a457ffde844ac391773170aad5d95a92L110), [link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-d57c4c1535dd3783fc94937211b66a786a621683678bcefdef775626c9e22c5eL125-R125), [link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-b21dcd67dbc0d24e0f8bba90701dfcfb5736cf40427547c7a7ccb91478f262cfL160-R160), [link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-69f811a90b3e91238aa9c2e34e8f18121a486f6630aaaf49d092682f60f01bc4), [link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-298b3426311d1616260c81704588becbb63bafaeca245d9b8480b53b684a6708))
*  Remove the `DefaultApplication` field from the `workspace` type and its assignments ([link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-9b6f6e9ffbcf999631c7d539beb30315bad491c95896f1569520a051b1d3c1d2L54-L56), [link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-bf4e917446c73565ee6967f28bdd8e619c0d4fa40b215f195b840c021dd36ce9L165-L168), [link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-bf4e917446c73565ee6967f28bdd8e619c0d4fa40b215f195b840c021dd36ce9L197-L200))
*  Update the test case for the `resource list` command to match the new error message ([link](https://github.com/radius-project/radius/pull/6837/files?diff=unified&w=0#diff-d57c4c1535dd3783fc94937211b66a786a621683678bcefdef775626c9e22c5eL125-R125))

### CODE REVIEW

File: pkg/cli/cmd/app/delete/delete_test.go [hunk 1]
```diff
@@ -44,7 +45,12 @@ func Test_Validate(t *testing.T) {
 			ExpectedValid: true,
 			ConfigHolder: framework.ConfigHolder{
 				ConfigFilePath: "",
-				Config:         radcli.LoadConfigWithWorkspaceAndApplication(t),
+				Config:         radcli.LoadConfigWithWorkspace(t),
+				DirectoryConfig: &config.DirectoryConfig{
+					Workspace: config.DirectoryWorkspaceConfig{
+						Application: "test-application",
+					},
+				},
```
Comment (rynowak) [inline pkg/cli/cmd/app/delete/delete_test.go:53]: This is configuring the test is basically the same way it was configured prior to the approval. 

We have both per-user and per-directory configuration now, so the per-user configuration for the app was redundant.

**Review by vinayada1 — APPROVED** [review]
Need to remove docs at: https://docs.radapp.io/reference/cli/rad_application_switch/

**Comment by rynowak** [comment]
> Need to remove docs at: https://docs.radapp.io/reference/cli/rad_application_switch/

Thanks for the reminder. These are generated docs, so it should be cleaned up automatically, but it doesn't hurt to check.

File: pkg/cli/clivalidation.go [hunk 2]
```diff
@@ -161,10 +161,6 @@ func RequireApplicationArgs(cmd *cobra.Command, args []string, workspace workspa
 		return "", err
 	}
 
-	if applicationName == "" {
-		applicationName = workspace.DefaultApplication
-	}
-
 	if applicationName == "" {
 		applicationName = workspace.DirectoryConfig.Workspace.Application
 	}
```
Comment (ytimocin) [inline pkg/cli/clivalidation.go:166]: Why was the first if deleted and not the second one?

**Review by ytimocin — APPROVED** [review]

File: pkg/cli/clivalidation.go (hunk 2 above)
Comment (rynowak) [reply pkg/cli/clivalidation.go:166]: As mentioned in the PR description we have two redundant ways for users to set the 'default application'. Line 164-167 is the per-user configuration file, which we're removing. Line 168-170 is the per-directory configuration file which will stay.

**Comment by rynowak** [comment]
> Need to remove docs at: https://docs.radapp.io/reference/cli/rad_application_switch/

FYI this worked :) https://github.com/radius-project/docs/pull/977

**Pull request merged**